docs/docs/administration/single-sign-on.md

```diff
@@ -0,0 +1,218 @@
+
+- `client_id`
+- `secret` (sometimes called `client secret`, `token`, or `client key`)
+
+These credentials authenticate Mathesar with your IdP.
+
+Add them to the provider block like so:
+
+```diff
+  provider1:
+    provider_name: okta
+    server_url: https://trial-example-admin.okta.com
++   client_id: 0oatafg35rDG2KVQD697
++   secret: 8xvA3s6pzl9cx7fit7LZ3RIZhAGgG9Rst509dijCVBXwKL3ijpjHbmPDPa0WXln1
+```
+
+You've now completed all the minimum requirements to enable Single Sign-On (SSO) in Mathesar. Next, you can:
+
+- Add additional providers, if needed, by repeating this process
+- Explore optional but **recommended** settings like [restricting access to particular email domains](#restrict-access-to-specific-email-domains) and [setting default database roles](#set-default-user-roles-for-your-databases)
+
+Finally, you'll need to restart Mathesar so it can read the `sso.yml` file and enable your changes. Be sure to  use the correct method for your installation:
+
+=== "For Docker Compose installations"
+
+    ```bash
+    docker compose down
+    docker compose up -d
+    ```
+
+=== "For Linux, macOS, or WSL installations"
+
+    ```bash
+    sudo systemctl restart mathesar.service
+    ```
+
+These options can help you tailor authentication and access control to better fit your organization's needs.
+
+## Additional configuration options
+
+You can extend the providers in your `sso.yml` file with additional settings to restrict access and define default roles for users. These fields are optional, but **highly recommended** for tightening security and streamlining user provisioning.
+
+### Restrict access to specific email domains
+
+Use the `allowed_email_domains` setting to restrict SSO logins to specific email domains. This is useful if your identity provider manages multiple domains or if you want to prevent unauthorized domains from accessing your Mathesar instance.
+
+- **Default:** No restriction (i.e., users from any domain can log in).
+- **Expected format:** A list of domain names (e.g., `['example.com', 'mathesar.org']`).
+
+**Example:**
+
+```diff
+oidc_providers:
+  provider1:
+    provider_name: okta
+    server_url: https://trial-example-admin.okta.com
+    client_id: YOUR_CLIENT_ID
+    secret: YOUR_SECRET
++   allowed_email_domains: ['example.com', 'mathesar.org']
+```
+
+With this configuration, only users whose email ends in `@example.com` or `@mathesar.org` will be allowed to log in.
+
+---
+
+### Set default user roles for your databases
+
+The `default_pg_role` block allows you to **automatically assign PostgreSQL roles** to users the first time they log in via SSO. You can configure this for one or more databases.
+
+Each database block must include the following:
+
+- `name`: The **PostgreSQL database name**
+- `host`: The **hostname** or IP address of the database
+- `port`: The **port** on which the database is running
+- `role`: The **PostgreSQL role** to assign to users
```

Comment:
Consider adding a screenshot for finding these values:

<img width="2008" height="1422" alt="Screenshot 2025-07-17 at 7 32 26 PM" src="https://github.com/user-attachments/assets/420282aa-c0ac-4bcd-9959-bad3568dba2c" />

<img width="2246" height="578" alt="Screenshot 2025-07-17 at 7 34 39 PM" src="https://github.com/user-attachments/assets/b4b92c9b-4254-443e-9952-9730ce9f0318" />

Reply:
Awesome idea, will do!